### PR TITLE
Raster bbox

### PR DESCRIPTION
**Pull request by dorukozturk**
2 things to note here:

1) Footprint is the boundary with no data areas masked as opposed to boundary box which is the extent of an image. See the comparison below. Red one is the bounding box and green one is the footprint. 

bbox_vs_footprint

2) Rasterio relies on raster to have a nodata tag. When gdalinfo is called on the layer like:

```
gdalinfo some_crazy_raster.tiff
```

We should see this:
```
Band 1 Block=387x10 Type=UInt16, ColorInterp=Gray
  Min=5970.000 Max=61100.000 
  Minimum=5970.000, Maximum=61100.000, Mean=8519.507, StdDev=2910.222
  NoData Value=0
  Metadata:
    STATISTICS_MAXIMUM=61100
    STATISTICS_MEAN=8519.5071550568
    STATISTICS_MINIMUM=5970
    STATISTICS_STDDEV=2910.2219569724
```

Note the NoData Value metadata. That value can be add with gdal_translate if missing. In the following case we are creating an output file with nodata value of -9999 attached.

```
gdal_translate -a_nodata -9999 input.tiff output.tiff
```

### CODE REVIEW

File: gaia/geo/gdal_functions.py [hunk 1]
```diff
@@ -445,6 +445,24 @@ def gdal_zonalstats(zones, raster):
...
+    """
+
+    raster = get_dataset(raster_input)
+    geotransform = raster.GetGeoTransform()
+    xmax = geotransform[0] + (raster.RasterXSize * geotransform[1])
+    ymin = geotransform[3] + (raster.RasterYSize * geotransform[5])
+    xmin = geotransform[0]
+    ymax = geotransform[3]
+    return [xmin, ymin, xmax, ymax]
+
```
Comment (danlipsa) [inline gaia/geo/gdal_functions.py]: Do you need to handle images that are not aligned with XY axes? In that case, I think you'll need to compute the coordinates for all corners and take the min/max along the X and Y axes.

Comment (dorukozturk) [reply gaia/geo/gdal_functions.py]: @danlipsa I added the get_footprint function which does what you've asked for.

Comment (danlipsa) [reply gaia/geo/gdal_functions.py]: For the parallelogram above (green), xmin for the bounding box (red) is not computed correctly isn't it?

Comment (dorukozturk) [reply gaia/geo/gdal_functions.py]: @danlipsa You are right. Rasterio's bounds method has the same problem. 
See https://github.com/mapbox/rasterio/issues/525. 

Footprint is computed correctly but bounding box is not for certain cases. I am trying to find a solution for this but not sure if there is a general solution for this problem.

**Comment by dorukozturk** [comment]
@danlipsa Computing the bounding box from the footprint was a good idea. There was already a function in shapely called "bounds" which was giving the bounding box ([see](https://toblerity.org/shapely/manual.html)).

I think this is ready for another round of review.

**Comment by danlipsa** [comment]
LGTM

**Pull request merged**